lib/inertia_rails/controller.rb

```diff
@@ -41,6 +67,14 @@ def default_render
       end
     end
 
+    def shared_data
+      # Theses variables are set in the before_action, but we need to ensure they are initialized when using renderer alone
+      @_inertia_shared_plain_data ||= {}
+      @_inertia_shared_blocks ||= []
+
+      @_inertia_shared_plain_data.merge(evaluated_blocks)
```

Comment:
This is a nitpick, but I'd prefer not to set instance variables inside a reader-like method when the method has a different name than the instance variable.

I'd prefer one of two options:

Create explicit reader methods that hold the defaults (and change the references in the controller to use this method instead of accessing the instance variable directly)

```ruby
module InertiaRails
  module Controller
    def inertia_shared_plain_data
      @_inertia_shared_plain_data ||= {}
    end

    def inertia_shared_blocks
      @_inertia_shared_blocks ||= []
    end
  end
end
```

or, just put the defaults inline in the reader methods:

```ruby
module InertiaRails
  module Controller
    def shared_data
      (@_inertia_shared_plain_data || {}).merge(evaluated_blocks)
    end

    def evaluated_blocks
      def evaluated_blocks
        (@_inertia_shared_blocks || []).map { |block| instance_exec(&block) }&.reduce(&:merge) || {}
      end
    end
  end
end
```
 
I don't have a strong opinion on which one.